core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala

```diff
@@ -44,8 +46,13 @@ trait SynapseMLLogging extends Logging {
 
   val uid: String
 
-  protected def logBase(methodName: String): Unit = {
-    logBase(SynapseMLLogInfo(uid, getClass.toString, methodName, BuildInfo.version))
+  protected def logBase(methodName: String, columns:Int = -1): Unit = {
+    logBase(SynapseMLLogInfo(
+      uid,
+      getClass.toString,
+      methodName,
+      BuildInfo.version,
+      if (columns == -1) None else Some(columns)))
```

Comment:
Maybe this can just take in an Option of int too because it isnt used much

Reply:
Did that. Please check.